src/foam/u2/crunch/CrunchController.js

```diff
@@ -56,27 +57,33 @@ foam.CLASS({
         ).select().then((results) => {
           // Get the Capability objects in a map because it's faster than sorting
           var capabilityMap = {};
-          results.array.forEach((cap) => { capabilityMap[cap.id] = cap; });
+          results.array.forEach((cap) => capabilityMap[cap.id] = cap);
+            // Collect lists for fullfilling the requirements for any given capability.
+            ofList = tcList
+              .filter((capID) => capabilityMap[capID].of != null)
+              .map((capID) => capabilityMap[capID].of);
 
-          // Construct a list of 'of' values from the list of Capability objects
-          // in the order of capability IDs resulting from the pre-order traversial.
-          ofList = tcList
-            .filter((capID) => capabilityMap[capID].of != null)
-            .map((capID) => capabilityMap[capID].of);
-        });
-      });
-      this.capabilityDAO.find(capabilityId).then((cap) => {
-        // Summon the wizard; accio!
-        p.then(() => {
-          let argsList = ofList.map(() => { return {}; });
-          self.stack.push({
-            class: 'foam.nanos.crunch.ui.ScrollSectionWizardView',
-            title: cap.name,
-            ofList: ofList,
-            argsList: argsList
+            daoList = tcList
+              .filter((capID) => capabilityMap[capID].of != null)
+              .map((capID) => capabilityMap[capID].daoKey);
+
+            argsList = tcList
+              .filter((capID) => capabilityMap[capID].of != null)
+              .map((capID) => this.ctrl[capabilityMap[capID].daoFindKey].id);
```

Comment:
```suggestion
              .map((capID) => {
                let keyObj = capabilityMap[capID].daoFindKey;
                if ( typeOf keyObj === 'Array' ) {
                  let newKeyObj = {};
                  keyObj.forEach((k, i) => {
                     newKeyObj.push(this.ctrl[k].id);
                  });
                  return newKeyObj;
                }
                return this.ctrl[capabilityMap[capID].daoFindKey].id);
              });
```
this might be a multipart key ... {user,agent} ... need to account for.